sonar-scanner-engine/src/main/java/org/sonar/scanner/genericcoverage/GenericTestExecutionReportParser.java

```diff
@@ -0,0 +1,159 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.scanner.genericcoverage;
+
+import com.google.common.base.Preconditions;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.xml.stream.XMLStreamException;
+import org.codehaus.staxmate.in.SMHierarchicCursor;
+import org.codehaus.staxmate.in.SMInputCursor;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.test.MutableTestCase;
+import org.sonar.api.test.MutableTestPlan;
+import org.sonar.api.test.TestCase;
+import org.sonar.api.utils.StaxParser;
+import org.sonar.scanner.deprecated.test.TestPlanBuilder;
+
+import static org.sonar.scanner.genericcoverage.GenericCoverageReportParser.checkElementName;
+import static org.sonar.scanner.genericcoverage.GenericCoverageReportParser.longValue;
+import static org.sonar.scanner.genericcoverage.GenericCoverageReportParser.mandatoryAttribute;
+
+public class GenericTestExecutionReportParser {
+
+  private static final String NAME_ATTR = "name";
+  private static final String DURATION_ATTR = "duration";
+  private static final String MESSAGE_ATTR = "message";
+  public static final String OK = "ok";
+  public static final String ERROR = "error";
+  public static final String FAILURE = "failure";
+  public static final String SKIPPED = "skipped";
+
+  private static final int MAX_STORED_UNKNOWN_FILE_PATHS = 5;
+
+  private final TestPlanBuilder testPlanBuilder;
+
+  private int numberOfUnknownFiles;
+  private final List<String> firstUnknownFiles = new ArrayList<>();
+  private final Set<String> matchedFileKeys = new HashSet<>();
+
+  public GenericTestExecutionReportParser(TestPlanBuilder testPlanBuilder) {
+    this.testPlanBuilder = testPlanBuilder;
+  }
+
+  public void parse(java.io.File reportFile, SensorContext context) {
+    try (InputStream inputStream = new FileInputStream(reportFile)) {
+      parse(inputStream, context);
+    } catch (Exception e) {
+      throw new IllegalStateException("Error during parsing of test execution report " + reportFile, e);
+    }
+  }
+
+  public void parse(InputStream inputStream, SensorContext context) throws XMLStreamException {
+    new StaxParser(rootCursor -> {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "StaxParser"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)